account-kit/smart-contracts/src/msca/plugins/session-key/utils.ts

```diff
@@ -6,24 +6,45 @@ import { AccountNotFoundError } from "@alchemy/aa-core";
 import type { Address, Chain, Client, Transport } from "viem";
 import { SessionKeyPlugin } from "./plugin.js";
 
-// find predecessors for each keys and returned the struct `ISessionKeyPlugin.SessionKeyToRemove[]`
-// where SessionKeyToRemove = { sessionKey: Address, predecessor: Hex }
-export const buildSessionKeysToRemoveStruct: <
+type BuildSessionKeysToRemoveStructParams<
+  TAccount extends SmartContractAccount | undefined =
+    | SmartContractAccount
+    | undefined
+> = {
+  keys: ReadonlyArray<Address>;
+  pluginAddress?: Address;
+} & GetAccountParameter<TAccount>;
+
+/**
+ * Finds predecessors for each provided key and returns them in the struct `ISessionKeyPlugin.SessionKeyToRemove[]`.
+ *
+ * @example
+ * ```ts
+ * import { buildSessionKeysToRemoveStruct } from "@account-kit/smart-contracts";
+ *
+ * const client = createSmartAccountClient(...);
+ *
+ * const keysToRemove = await buildSessionKeysToRemoveStruct(client, {
+ *  keys: ["0x...", "0x..."],
+ * });
+ * ```
+ *
+ * @param {Client<TTransport, TChain, TAccount>} client The client instance used to interact with the smart contract
```

Comment:
```suggestion
 * @param {Client<TTransport, TChain, TAccount>} client The client instance used to interact with the smart account
```

not 100% positive this is meant to refer to the account here